contract/models/abstract_contract_line.py

```diff
@@ -224,12 +224,17 @@ def _inverse_price_unit(self):
         for line in self.filtered(lambda x: not x.automatic_price):
             line.specific_price = line.price_unit
 
+    def _compute_price_subtotal_helper(self):
+        self.ensure_one()
+        subtotal = self.quantity * self.price_unit
+        discount = self.discount / 100
+        subtotal *= 1 - discount
+        return subtotal
+
     @api.depends("quantity", "price_unit", "discount")
     def _compute_price_subtotal(self):
         for line in self:
-            subtotal = line.quantity * line.price_unit
-            discount = line.discount / 100
-            subtotal *= 1 - discount
+            subtotal = line._compute_price_subtotal_helper()
```

Comment:
@barkat-matthias Thanks for this.

For a better comprehension reading the code, it is better to not name the hook function `_compute_...` 

I'd prefer `_get_price_subtotal()`